system-tests/_support/index.js

```diff
@@ -0,0 +1,188 @@
+  });
+
+  // If nothing found, return empty selection
+  expect(formGroup).not.to.equal(null);
+
+  // If we found a form group, return the input element within
+  // (Note: <span class="form control"> is used to wrap <select>s)
+  return formGroup.find('textarea, input, select, *:not(span).form-control');
+});
+
+/**
+ * Locate a table row that contains the given string and return the entire row
+ *
+ * This function searches all the <tr /> elements within the current scope and
+ * checks if it contains the string given. If matched, the scope is filtered
+ * to the <tr /> element(s) found.
+ *
+ * @param {jQuery.Element} elements - The DOM scope to search within
+ * @param {String} contents - The contents of the <td /> to search
+ */
+Cypress.addChildCommand('getTableRowThatContains', function (elements, contents) {
+  let matchedRows = elements.find('tr');
+
+  matchedRows = matchedRows.filter(function (index, row) {
+    return Array.prototype.slice.apply(row.childNodes).some(function (child) {
+      return child.innerText.indexOf(contents) !== -1;
+    });
+  });
+
+  return Cypress.$(matchedRows);
+});
+
+/**
+ * Return all the <td> cells for the given column of the table
+ *
+ * This function collects all the <td /> cells for the given column (addressed
+ * either by it's index, or by it's <th> contents).
+ *
+ * @param {jQuery.Element} elements - The DOM scope to search within
+ * @param {String|Number} columNameOrIndex - The table index or the <th /> contents
+ */
+Cypress.addChildCommand('getTableColumn', function (elements, columNameOrIndex) {
+  const matchedRows = elements.find('tr');
+  const headings = matchedRows.eq(0).find('th');
+  let columnIndex = parseInt(columNameOrIndex);
+
+  if (Number.isNaN(columnIndex)) {
+    const compareName = String(columNameOrIndex).toLowerCase();
+
+    // Fallback to first column
+    columnIndex = 0;
+    headings.each(function (index, th) {
+      if (StringUtil.getContents(th).toLowerCase() === compareName) {
+        columnIndex = index;
+      }
+    });
+  }
+
+  // Collect all visible rows
+  return matchedRows
+    .slice(1)
+    .filter(function (index, element) {
+      return Cypress.$(element).is(':visible');
+    })
+    .map(function (index, element) {
+      return element.querySelectorAll('td')[columnIndex];
+    });
+});
+
+/**
+ * Return the text contents of all the selected elements
+ *
+ * This function will return an array with the text contents of every element
+ * in the context. This function extracts the contents like so:
+ *
+ * - .ace_editor : Uses ACE Editor API to get contents
+ * - input       : Returns the value property
+ * - *           : Returns the innerText value
+ *
+ * @param {jQuery.Element} elements - The DOM scope to search within
+ */
+Cypress.addChildCommand('contents', function (elements) {
+  return elements.map(function (index, element) {
+    const doc = element.ownerDocument;
+    const win = doc.defaultView || doc.parentWindow;
```

Comment:
Do we use short words not to clash with `window` and `document`? Do they exist in this context?